carp.protocols.core/src/commonTest/kotlin/dk/cachet/carp/protocols/domain/deployment/UnexpectedMeasuresWarningTest.kt

```diff
@@ -0,0 +1,50 @@
+package dk.cachet.carp.protocols.domain.deployment
+
+import dk.cachet.carp.common.data.DataType
+import dk.cachet.carp.protocols.infrastructure.test.StubMasterDeviceDescriptor
+import dk.cachet.carp.protocols.infrastructure.test.StubMeasure
+import dk.cachet.carp.protocols.infrastructure.test.StubTaskDescriptor
+import dk.cachet.carp.protocols.infrastructure.test.createEmptyProtocol
+import kotlin.test.*
+
+
+class UnexpectedMeasuresWarningTest
+{
+    @Test
+    fun isIssuePresent_true_when_unexpected_measure_in_protocol()
+    {
+        val protocol = createEmptyProtocol()
+        val master = StubMasterDeviceDescriptor()
+        protocol.addMasterDevice( master )
+        val expectedMeasure = StubMeasure()
+        val unexpectedMeasure = StubMeasure( DataType( "namespace", "unexpected" ) )
```

Comment:
How come `unexpectedMeasure` is unexpected? Is it because it contains a `DataType`?

Reply:
Because 'StubMasterDeviceDescriptor.supportedDataTypes' is empty. Could you update the test to make it clearer to the reader what measures the `StubMasterDeviceDescriptor()` supports and doesn't? Maybe pass the `expectedMeasure` into `StubMasterDeviceDescriptor()`?

Reply:
The `DataType` within the measure is not listed in the supported data types of the device. `StubmasterDeviceDescriptor.supportedDataTypes` is not empty, it contains `STUB_DATA_TYPE`.

That is hidden away so I understand that is unclear. Think I will indeed allow caller to pass supported types to the stubs.

Reply:
Fixed in https://github.com/cph-cachet/carp.core-kotlin/pull/165/commits/51bd4cb0212cad2a03c97485d3157fbd47da6e11